src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -136,6 +141,45 @@ internal JsonClassInfo(Type type, JsonSerializerOptions options)
             }
         }
 
+        private JsonPropertyInfo GetPropertyThatHasAttribute<TAttribute>() where TAttribute : Attribute
+        {
+            JsonPropertyInfo property = null;
+
+            for (int iProperty = 0; iProperty < _propertyRefs.Count; iProperty++)
+            {
+                PropertyRef propertyRef = _propertyRefs[iProperty];
+                JsonPropertyInfo jsonPropertyInfo = propertyRef.Info;
+                TAttribute attribute = jsonPropertyInfo.PropertyInfo.GetCustomAttribute<TAttribute>(inherit: true);
+                if (attribute != null)
+                {
+                    if (property != null)
+                    {
+                        ThrowHelper.ThrowInvalidOperationException_SerializationDuplicateAttribute(typeof(TAttribute));
+                    }
+
+                    property = jsonPropertyInfo;
+                }
+            }
+
+            return property;
+        }
+
+        private void DetermineExtensionDataProperty()
+        {
+            JsonPropertyInfo jsonPropertyInfo = GetPropertyThatHasAttribute<JsonExtensionDataAttribute>();
+            if (jsonPropertyInfo != null)
+            {
+                Type declaredPropertyType = jsonPropertyInfo.DeclaredPropertyType;
+                if (!declaredPropertyType.IsAssignableFrom(typeof(Dictionary<string, JsonElement>)) &&
```

Comment:
Is there any reason not to support `IDictionary<TKey, TValue>` here? A lot of devs like to expose collections on there models as an interfaces

Reply:
We should support it, yes. Thanks. It was an oversight - we support it in all other cases and there is not valid reason to lock it down to just `Dictionary`